File: solutions/python/armstrong-numbers/4/armstrong_numbers.py
def is_armstrong_number(number):
    """
    Determine if a given number is an Armstrong number.

    An Armstrong number (also known as a narcissistic number) is a number that is equal to the sum of its own digits raised to the power of the number of digits. For example,
    153 is an Armstrong number because 1^3 + 5^3 + 3^3 = 153.

    Args:
        number (int): The number to check.

    Returns:
        bool: True if the number is an Armstrong number, False otherwise.
    """
    string_number = str(number)
    string_length = len(string_number)
    return sum(int(digit) ** string_length for digit in string_number) == number

File: solutions/javascript/armstrong-numbers/2/armstrong-numbers.js
/**
 * Checks if a given number is an Armstrong number.
 *
 * An Armstrong number is a number that is equal to the sum of its own digits each raised to the power of the number of digits.
 *
 * @param {number|string} input - The number to check.
 * @returns {boolean} True if the input is an Armstrong number, false otherwise.
 */
export const isArmstrongNumber = (input) => {
  const bigInput = BigInt(input);

  // Convert the number to a string and split it into individual digits
  const digits = [...String(bigInput)];

  // Calculate the sum of each digit raised to the power of the number of digits
  const sum = digits.reduce(
    (total, current) => total + BigInt(current) ** BigInt(digits.length),
    BigInt(0)
  );

  // Check if the calculated sum is equal to the original number
  return sum === bigInput;
};
